src/pudl/transform/vcerare.py

```diff
@@ -0,0 +1,330 @@
+                "lat_county": "latitude",
+                "long_county": "longitude",
+                "subdivision_code": "state",
+            }
+        )
+        .drop(columns=["state_id_fips", "fips"])
+    )
+    return lat_long_fips
+
+
+def _add_time_cols(df: pd.DataFrame, df_name: str) -> pd.DataFrame:
+    """Add datetime and hour_of_year columns.
+
+    This function adds a datetime column and a hour_of_year column.
+    The datetime column is important for merging the data with other
+    data, and the hour_of_year 1-8760 column is important for modeling
+    purposes. The report_year column is also helpful for filtering,
+    so we keep all three!
+
+    For leap years (2020), December 31st is excluded.
+    """
+    logger.info(f"Adding time columns for {df_name} table")
+    # This data is compiled for modeling purposes and skips the last
+    # day of a leap year. When adding a datetime column, we need
+    # to make sure that we skip the 31st of December, 2020 and that
+    # every year has exactly 8760 hours in it.
+    all_years = df.report_year.unique()
+    datetime8760_index = pd.DatetimeIndex(
+        pd.concat(
+            [
+                pd.Series(pd.date_range(start=f"{year}-01-01", periods=8760, freq="h"))
+                for year in all_years
+            ]
+        )
+    )
+    new_time_col = pd.DataFrame(
+        {
+            "datetime_utc": datetime8760_index,
+        }
+    )
+    df = pd.concat(
+        [df.reset_index(drop=True), new_time_col.reset_index(drop=True)], axis="columns"
+    ).rename(columns={"unnamed_0": "hour_of_year"})
+    return df
+
+
+def _stack_cap_fac_df(df: pd.DataFrame, df_name: str) -> pd.DataFrame:
+    """Funciton to transform each capacity factor table individually to save memory.
+
+    The main transforms are turning count/subregion columns into county/subregion rows
+    and renaming columns to be more human-readable and compatible with the FIPs df
+    that will get merged in.
+
+    This function is intended to save memory by being applied to each individual
+    capacity factor table rather than the giant combined one.
+    """
+    logger.info(f"Stacking the county/subregion columns for {df_name} table.")
+    df_stacked = (
+        df.set_index(["datetime_utc", "hour_of_year", "report_year"])
+        .stack()
+        .reset_index()
+        .rename(
+            columns={"level_3": "county_state_names", 0: f"capacity_factor_{df_name}"}
+        )
+        .assign(county_state_names=lambda x: pd.Categorical(x.county_state_names))
+    )
+    return df_stacked
+
+
+def _make_cap_fac_frac(df: pd.DataFrame, df_name: str) -> pd.DataFrame:
+    """Make the capacity factor column a fraction instead of a percentage.
+
+    This step happens before the table gets stacked to save memory.
+    """
+    logger.info(f"Converting capacity factor into a fraction for {df_name} table.")
+    county_cols = [x for x in df.columns if x not in ["report_year", "unnamed_0"]]
```

Comment:
Non-blocking: It might be clearer to rename `unnamed_0` before you do any of these steps, and then reference it directly so it's more obvious why it's getting excluded here.